src/Nncase.Transform/Rules/Neutral/CombineTranspose.cs

```diff
@@ -61,7 +61,7 @@ public sealed partial class CombineTransposeConstBinary : IRewriteRule
     public CombineTransposeConstBinary()
     {
         var perm = IsWildcard("perm");
-        Pattern = IsAlt(IsBinary("binary", _ => true, IsTranspose(IsWildcard("x"), perm), IsConst("y")), IsBinary("binary", _ => true, IsConst("x"), IsTranspose(IsWildcard("y"), perm)));
+        Pattern = IsAlt(IsBinary("binary", _ => true, IsTranspose(IsWildcard("x"), perm), IsConst("y", y => y.CheckedShape.Count == 1)), IsBinary("binary", _ => true, IsConst("x", x => x.CheckedShape.Count == 1), IsTranspose(IsWildcard("y"), perm)));
```

Comment:
It's more semantically meaningful to use `IsConst("y") with { TypePattern = HasRank(1) }`.

Reply:
resolved